_posts/2019-01-31-CocoaPods-1.7.0-beta.markdown

```diff
@@ -0,0 +1,121 @@
+
+However, there are many cases in which consumers of a pod might be unable to use the latest Swift version, perhaps because their toolchain does not support it yet. For example, a project that is using an older version of Xcode that does not support Swift 4 will be unable to integrate the `CoconutLib` pod. To solve this, CocoaPods 1.7.0 provides the ability to specify Swift version requirements for each target that is integrated within the `Podfile`.
+
+For example:
+
+```ruby
+target 'MyApp' do
+  supports_swift_versions '>= 3.0', '< 4.0'
+  pod 'CoconutLib', '~> 1.0'
+end
+```
+
+This will successfully integrate `CoconutLib` pod but instead use Swift 3 as the version because that is the only version satisfied by the requirements specified by the `MyApp` and by the list of supported versions of Swift from the `CoconutLib` pod.
+
+**Note**: `supports_swift_versions` can also be declared at the root level of a `Podfile` in which case it will apply to all targets within the `Podfile`. Furthermore, nested targets (such as test targets) will inherit their Swift requirements from their parent.
+
+There are many other edge cases that can occur during this integration. We encourage you to read through the [proposal](https://github.com/CocoaPods/CocoaPods/issues/8191) for this change to find out more!
+
+#### Linting And Validation
+
+It is generally difficult for pod authors to maintain infrastructure to support multiple (primarily older) versions of Swift, therefore, during `lint` CocoaPods will choose the _latest_ version of Swift.
+
+For pod authors who do have the infrastructure (such as a CI system) that ensure their pod works with older versions of Swift, the `--swift-version` parameter should be used to override the default behavior.
+
+#### Deprecating `.swift-version` File
+
+Most pod authors have been relying on specifying a `.swift-version` file at the root of their repo in order to succesfully publish their pod with the Swift version they officially support. However, this information is _never_ transcribed within the `podspec` and therefore during integration none of it is being used when choosing which Swift version to use.
+
+This causes numerous issues especially when a new version of Swift is released. This is because a consumer will automatically pick the latest version of Swift for compilation without the pod author officially claiming that it is supported.
+
+**We strongly recommend pod authors to delete the `.swift-version` file and instead migrate over to use the officially supported `swift_version` DSL within their `podspec` instead.**
+
+A warning will be displayed during `lint` that encourages pod authors to migrate away from using the `.swift-version` file and in a future major release of CocoaPods we plan to completely remove support for it.
+
+### App Specs
+
+TODO
+
+### Multiple Xcodeproj Generation
+
+TODO
+
+### Incremental Installation
+
+TODO
+
+### CDN Support For Master Repo
+
+TODO
```

Comment:
@igor-makarov I'd love if you can type up something here for the blog post of 1.7.0 beta.

Do not worry we are not ready for a beta yet but I figured I could start the blog post.

Reply:
\### CDN Support For Master Repo
\*\*EXPERIMENTAL**
Use \`source 'https://cdn.jsdelivr.net/cocoa/'` in your Podfile to avoid cloning the main spec repo on your machine and especially on CI.

Reply:
thanks I will add a bit more context to it :)

Reply:
@igor-makarov I tried that in a sample app and I am unable to install `Alamofire` for example. Am I missing something? The repo seems to be updated.